File: mission/js/elementary/three-words.js
"use strict";

function threeWords(data) {
    let dataArray = data.split(" ");
    var count = 0;
    var returnResult = false;
    dataArray.forEach(function (item) {
        if (isNaN(Number(item))) {
            count++;
            if (count === 3) {
                returnResult = true;
            }
        } else {
            count = 0;
        }
    });
    return returnResult;
}

var assert = require('assert');

if (!global.is_checking) {
    assert.equal(threeWords("Hello World hello"), true, "1st example");
    assert.equal(threeWords("He is 123 man"), false, "2nd example");
    assert.equal(threeWords("1 2 3 4"), false, "3rd example");
    assert.equal(threeWords("bla bla bla bla"), true, "4th example");
    assert.equal(threeWords("Hi"), false, "Letters");
    console.log("Coding complete? Click 'Check' to review your tests and earn cool rewards!");
}

File: mission/python/elementary/three-words.py
import re


def checkio(words):
    words_list = words.split(' ')
    count = 0
    for word in words_list:
        if re.match('[a-zA-Z]+', word):
            count = count + 1
            if count == 3:
                return True
        else:
            count = 0
    return False


if __name__ == '__main__':
    assert checkio("Hello World hello") == True, "Hello"
    assert checkio("He is 123 man") == False, "123 man"
    assert checkio("1 2 3 4") == False, "Digits"
    assert checkio("bla bla bla bla") == True, "Bla Bla"
    assert checkio("Hi") == False, "Hi"
    print("Coding complete? Click 'Check' to review your tests and earn cool rewards!")
